File: ReadSys.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <ctime>
#include <unistd.h>
#include <limits>
#include <map>

using namespace std;

// Función para leer archivos de /proc
string leerArchivoProc(const string& ruta) {
    ifstream archivo(ruta);
    if (!archivo.is_open()) {
        return "Error: No se pudo abrir " + ruta;
    }
    
    string contenido((istreambuf_iterator<char>(archivo)), 
                    istreambuf_iterator<char>());
    archivo.close();
    
    return contenido;
}

// Mostrar información de tiempo
void mostrarTiempo() {
    time_t ahora = time(nullptr);
    cout << "\nHora actual del sistema: " << ctime(&ahora);
    cout << "Tiempo en segundos desde epoch: " << ahora << endl;
}

// Mostrar información del stack
void mostrarStack() {
    string ruta_stack = "/proc/self/stack";
    cout << "\nInformación del stack (" << ruta_stack << "):\n";
    cout << leerArchivoProc(ruta_stack) << endl;
}

// Mostrar información de varios archivos en /proc
void mostrarInfoProc() {
    map<string, string> archivos_proc = {
        {"cpuinfo", "/proc/cpuinfo"},
        {"meminfo", "/proc/meminfo"},
        {"version", "/proc/version"},
        {"uptime", "/proc/uptime"},
        {"loadavg", "/proc/loadavg"}
    };
    
    cout << "\nInformación del sistema desde /proc:\n";
    for (const auto& [nombre, ruta] : archivos_proc) {
        cout << "\nContenido de " << nombre << " (" << ruta << "):\n";
        cout << leerArchivoProc(ruta) << endl;
    }
}

// Limpiar el buffer de entrada
void limpiarBuffer() {
    cin.clear();
    cin.ignore(numeric_limits<streamsize>::max(), '\n');
}

// Menú principal
int main() {
    cout << string(50, '=') << endl;
    cout << "Monitor de Sistema - Información desde /proc" << endl;
    cout << string(50, '=') << endl;
    
    while (true) {
        cout << "\nOpciones disponibles:\n";
        cout << "1. Mostrar hora del sistema\n";
        cout << "2. Mostrar información del stack\n";
        cout << "3. Mostrar información de /proc\n";
        cout << "4. Salir\n";
        
        cout << "Seleccione una opción (1-4): ";
        int opcion;
        cin >> opcion;
        limpiarBuffer();
        
        switch (opcion) {
            case 1:
                mostrarTiempo();
                break;
            case 2:
                mostrarStack();
                break;
            case 3:
                mostrarInfoProc();
                break;
            case 4:
                cout << "Saliendo del programa..." << endl;
                return 0;
            default:
                cout << "Opción no válida. Por favor seleccione 1-4." << endl;
        }
        
        cout << "\nPresione Enter para continuar...";
        limpiarBuffer();
    }
    
    return 0;
}

File: ReadSys.py
import os
import time
from datetime import datetime

def leer_archivo_proc(ruta):
    """Función para leer archivos de /proc y devolver su contenido"""
    try:
        with open(ruta, 'r') as archivo:
            contenido = archivo.read()
        return contenido
    except IOError as e:
        return f"Error al leer el archivo {ruta}: {str(e)}"

def mostrar_tiempo():
    """Muestra la hora actual del sistema"""
    ahora = datetime.now()
    print(f"\nHora actual del sistema: {ahora.strftime('%Y-%m-%d %H:%M:%S')}")
    print(f"Tiempo en segundos desde epoch: {time.time()}")

def mostrar_stack():
    """Muestra información del stack desde /proc/self/stack"""
    ruta_stack = "/proc/self/stack"
    print(f"\nInformación del stack ({ruta_stack}):")
    print(leer_archivo_proc(ruta_stack))

def mostrar_info_proc():
    """Muestra información de varios archivos en /proc"""
    archivos_proc = {
        "cpuinfo": "/proc/cpuinfo",
        "meminfo": "/proc/meminfo",
        "version": "/proc/version",
        "uptime": "/proc/uptime",
        "loadavg": "/proc/loadavg"
    }
    
    print("\nInformación del sistema desde /proc:")
    for nombre, ruta in archivos_proc.items():
        print(f"\nContenido de {nombre} ({ruta}):")
        print(leer_archivo_proc(ruta))

def main():
    print("="*50)
    print("Monitor de Sistema - Información desde /proc")
    print("="*50)
    
    while True:
        print("\nOpciones disponibles:")
        print("1. Mostrar hora del sistema")
        print("2. Mostrar información del stack")
        print("3. Mostrar información de /proc")
        print("4. Salir")
        
        opcion = input("Seleccione una opción (1-4): ")
        
        if opcion == "1":
            mostrar_tiempo()
        elif opcion == "2":
            mostrar_stack()
        elif opcion == "3":
            mostrar_info_proc()
        elif opcion == "4":
            print("Saliendo del programa...")
            break
        else:
            print("Opción no válida. Por favor seleccione 1-4.")
        
        input("\nPresione Enter para continuar...")

if __name__ == "__main__":
    main()